src/Microsoft.Health.Api/Features/HealthChecks/CachedHealthCheck.cs

```diff
@@ -139,7 +139,12 @@ private bool HasExpired(DateTimeOffset timestamp)
         => timestamp >= _cache.ExpireTime;
 
     private bool IsUpToDate(DateTimeOffset timestamp)
-        => timestamp < _cache.StaleTime;
+    {
+        // While we will store results whose status is < MinimumCachedHealthStatus, it will effectively
+        // have no caching as IsUpToDate will return false
+        CachedHealthCheckResult currentCache = _cache;
+        return timestamp < currentCache.StaleTime && currentCache.Result.Status >= _options.MinimumCachedHealthStatus;
```

Comment:
So if its unhealthy/degraded we always want to evaluate? I guess for most of the services this won't be an issue, but we have few services who are always unhealthy we may do lot of calls and can be throttled

Reply:
This is definitely true, and something we'll have to account for using this setting.